plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -5626,6 +5626,42 @@ public void testOptimizeCleansUpDeleteFiles()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testOptimizeFilesDoNotInheritSequenceNumber()
+            throws IOException
+    {
+        String tableName = "test_optimize_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM nation", 25);
+
+        assertUpdate("DELETE FROM " + tableName + " WHERE nationkey = 7", 1);
+
+        // Verify that delete file exists
+        assertQuery(
```

Comment:
`assertQuery` isn't recommended, though it isn't marked as deprecated. We could use `assertThat(query(...))` or `computeScalar` instead.